### PR TITLE
plugins/lsp/jsonnet: init

### LINKED ISSUE #1687: [LSP REQUEST] Jsonnet LSP
| Field    | Description |
|----------|-------------|
| Plugin   | `jsonnet_ls` in lspconfig  | 
| Homepage | https://github.com/grafana/jsonnet-language-server     | 
| Nixpkgs  | https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/language-servers/jsonnet-language-server/default.nix | 

## Extra Information

Should work like all the other lspconfig LSPs.

### PR DESCRIPTION
**Pull request by MattSturgeon**
Fixes #1687

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -248,6 +248,13 @@ let
       ];
       settings = cfg: { json = cfg; };
     }
+    {
+      name = "jsonnet";
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/default.nix:252]:
```suggestion
      name = "jsonnet-ls";
```
To be consistent with the `nvim-lspconfig` naming.

File: plugins/lsp/language-servers/jsonnet-settings.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#  - https://github.com/grafana/jsonnet-language-server/blob/main/pkg/server/configuration.go
+#  - https://github.com/google/go-jsonnet/blob/master/internal/formatter/jsonnetfmt.go#L55
+with lib;
+{
+  ext_vars = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
+    External variables.
+  '';
+
+  formatting = {
+    Indent = helpers.defaultNullOpts.mkInt 2 ''
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/jsonnet-settings.nix:13]:
```suggestion
    Indent = helpers.defaultNullOpts.mkUnsignedInt 2 ''
```

File: plugins/lsp/language-servers/default.nix (hunk 1 above)
Comment (MattSturgeon) [reply plugins/lsp/language-servers/default.nix]: Done

**Review by GaetanLepage — APPROVED** [review]
LGTM, thanks !

**Pull request merged**